app/src/main/java/agersant/polaris/PolarisPlaybackService.java

```diff
@@ -116,15 +159,59 @@ public void onReceive(Context context, Intent intent) {
 		autoSaveHandler.postDelayed(autoSaveRunnable, 5000);
 
 		pushSystemNotification();
+
+		mediaSession = new MediaSessionCompat(this, getPackageName());
+		mediaSession.setCallback(new Callback(player));
+        mediaSession.setActive(true);
+
+		updateMediaSessionState(PlaybackStateCompat.STATE_NONE);
+
+		dataUpdateRunnable = () -> {
+			updateMediaSessionState(player.isPlaying() ? PlaybackStateCompat.STATE_PLAYING : PlaybackStateCompat.STATE_PAUSED);
+			dataUpdateHandler.postDelayed(dataUpdateRunnable, 5000);
+		};
+		dataUpdateHandler = new Handler();
+		startDataUpdates();
+
+	}
+
+	private void updateMediaSessionState(int state) {
+		PlaybackStateCompat.Builder builder = new PlaybackStateCompat.Builder();
+		builder.setActions(PlaybackStateCompat.ACTION_PLAY | PlaybackStateCompat.ACTION_PLAY_PAUSE | PlaybackStateCompat.ACTION_PAUSE | PlaybackStateCompat.ACTION_SKIP_TO_NEXT | PlaybackStateCompat.ACTION_SKIP_TO_PREVIOUS);
+		builder.setState(state, (long)player.getCurrentPosition(), 1);
+		mediaSession.setPlaybackState(builder.build());
+
+		MediaMetadataCompat.Builder metadataBuilder = new MediaMetadataCompat.Builder();
+		if(player.getCurrentItem() != null) {
+			metadataBuilder.putString(MediaMetadataCompat.METADATA_KEY_TITLE, player.getCurrentItem().getTitle());
+			metadataBuilder.putString(MediaMetadataCompat.METADATA_KEY_ARTIST, player.getCurrentItem().getArtist());
+			metadataBuilder.putString(MediaMetadataCompat.METADATA_KEY_ALBUM, player.getCurrentItem().getAlbum());
+			metadataBuilder.putString(MediaMetadataCompat.METADATA_KEY_ALBUM_ARTIST, player.getCurrentItem().getAlbumArtist());
+			metadataBuilder.putLong(MediaMetadataCompat.METADATA_KEY_TRACK_NUMBER, player.getCurrentItem().getTrackNumber());
+			metadataBuilder.putLong(MediaMetadataCompat.METADATA_KEY_DISC_NUMBER, player.getCurrentItem().getDiscNumber());
+		}
+		metadataBuilder.putLong(MediaMetadataCompat.METADATA_KEY_DURATION, (long)player.getDuration());
+		mediaSession.setMetadata(metadataBuilder.build());
 	}
 
 	@Override
 	public void onDestroy() {
+	    mediaSession.release();
 		unregisterReceiver(receiver);
 		autoSaveHandler.removeCallbacksAndMessages(null);
+		stopDataUpdates();
 		super.onDestroy();
 	}
 
+	private void startDataUpdates() {
```

Comment:
Similar to the runnable/handler, these should have names involving MediaSession.